src/java.base/share/classes/java/util/concurrent/ConcurrentHashMap.java

```diff
@@ -3742,8 +3742,8 @@ public void forEach(long parallelismThreshold,
      * @param <U> the return type of the transformer
```

Comment:
I guess we could discuss how to interpret this. `U` after this change is not the return type of transformer, but it never was - it was `? extends U`...
The alternative would be to keep U nullable and change the `action` to `? super @NonNull U`.
Given the null-hostility of `ConcurrentHashMap`, I like the PR, which highlights the allowed nulls more explicitly.